.github/workflows/build-wheels.yml

```
@@ -31,7 +31,32 @@ jobs:
   macos-test:
     runs-on: macos-13
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout LLVM on macOS
+        uses: actions/checkout@v4
+        with:
+          repository: llvm/llvm-project
+          ref: release/18.x
+          path: llvm-project
+
+      - name: Build OpenMP on macOS
+        run: |
+          cd llvm-project
+          cmake \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_INSTALL_PREFIX=$(brew --prefix) \
+            -DCMAKE_C_COMPILER=clang \
+            -DCMAKE_CXX_COMPILER=clang++ \
+            -DLIBOMP_INSTALL_ALIASES=OFF \
+            -S openmp \
+            -B build
+          cmake \
+            --build build \
+            --parallel
+          cmake --install build
+          cd "$GITHUB_WORKSPACE"
```

```suggestion
      - name: Build OpenMP on macOS
        env:
          MACOSX_DEPLOYMENT_TARGET: "10.9"
        working-directory: llvm-project
        run: |
          cmake \
            -DCMAKE_BUILD_TYPE=Release \
            -DCMAKE_INSTALL_PREFIX=$(brew --prefix) \
            -DCMAKE_C_COMPILER=clang \
            -DCMAKE_CXX_COMPILER=clang++ \
            -DLIBOMP_INSTALL_ALIASES=OFF \
            -S openmp \
            -B build
          cmake --build build --parallel
          cmake --install build
```

Some additional improvements from @henryiii (https://github.com/thaler-lab/Wasserstein/issues/28#issuecomment-2178283871) RE: `MACOSX_DEPLOYMENT_TARGET`.